tests/components/bayesian/test_config_flow.py

```diff
@@ -0,0 +1,1540 @@
+                CONF_ENTITY_ID: "sensor.office_illuminance_lux",
+                CONF_ABOVE: 40,
+                CONF_P_GIVEN_T: 85,
+                CONF_P_GIVEN_F: 45,
+                CONF_NAME: "Office is bright",
+                "add_another": True,
+            },
+        )
+        await hass.async_block_till_done()
+
+        # Since "add_another" was True, we should return to the observation selector
+        assert result3["step_id"] == str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+        assert result3["type"] is FlowResultType.MENU
+
+        # Add a state observation
+        result4 = await hass.config_entries.flow.async_configure(
+            result3["flow_id"], {"next_step_id": str(ObservationTypes.STATE)}
+        )
+        await hass.async_block_till_done()
+
+        assert result4["step_id"] == str(ObservationTypes.STATE)
+        assert result4["type"] is FlowResultType.FORM
+        result5 = await hass.config_entries.flow.async_configure(
+            result4["flow_id"],
+            {
+                CONF_ENTITY_ID: "sensor.work_laptop",
+                CONF_TO_STATE: "on",
+                CONF_P_GIVEN_T: 60,
+                CONF_P_GIVEN_F: 20,
+                CONF_NAME: "Work laptop on network",
+                "add_another": True,
+            },
+        )
+        await hass.async_block_till_done()
+
+        # Since "add_another" was True, we should return to the observation selector
+        assert result5["step_id"] == str(ConfigFlowSteps.OBSERVATION_SELECTOR)
+        assert result5["type"] is FlowResultType.MENU
+
+        # Lastly, add a template observation
+        result6 = await hass.config_entries.flow.async_configure(
+            result5["flow_id"], {"next_step_id": str(ObservationTypes.TEMPLATE)}
+        )
+        await hass.async_block_till_done()
+
+        assert result6["step_id"] == str(ObservationTypes.TEMPLATE)
+        assert result6["type"] is FlowResultType.FORM
+        result7 = await hass.config_entries.flow.async_configure(
+            result6["flow_id"],
+            {
+                CONF_VALUE_TEMPLATE: """
+{% set current_time = now().time() %}
+{% set start_time = strptime("07:00", "%H:%M").time() %}
+{% set end_time = strptime("18:30", "%H:%M").time() %}
+{% if start_time <= current_time <= end_time %}
+True
+{% else %}
+False
+{% endif %}
+                """,
+                CONF_P_GIVEN_T: 45,
+                CONF_P_GIVEN_F: 5,
+                CONF_NAME: "Daylight hours",
+                "add_another": False,
+            },
+        )
+
+        # Since add_another is false, we should now be done.
+        assert result7["version"] == 1
+        assert result7["options"] == {
+            CONF_NAME: "Office occupied",
+            CONF_PROBABILITY_THRESHOLD: 0.5,
+            CONF_PRIOR: 0.15,
+            CONF_DEVICE_CLASS: "occupancy",
+            CONF_OBSERVATIONS: [
+                {
+                    CONF_PLATFORM: str(ObservationTypes.NUMERIC_STATE),
```

Comment:
I think this should be `CONF_TYPE` or something. It got me very confused as it's not a platform

Reply:
This schema naming structure was copied from the original YAML config structure and naming convention (which I was not responsible for)
Options seem to be:
1)  Have a different schema for config entries and YAML - would require so ugly code in `binary_sensor.py` to handle this
2) Rename it across both YAML and config entries - breaking change unless we created some backwards compatibility (as I don't think we are meant to migrate people's YAML for them)
3) Comment the code to prevent future confusion

My preference is for 3.
We could think about 2 in a separate PR with a deprecation period and then migrations and repair notifications.

Reply:
There are no reasons why the schema in the config flow needs to be exactly as in the platform setup (while of course to avoid confusion it's good to keep it consistent).
I suggest you use better key's here that are suitable.